### DISCUSSION TITLE
avrcp volume control

### DISCUSSION THREAD

**Discussion by AvishkaDon** [pschatzmann/ESP32-A2DP#181] [Q&A] [answered]
Hi,

I was wondering if the `BluetoothA2DPSink::set_volume(uint8_t volume)` function set the absolute volume? (i.e when I change the volume on the esp32 the volume slider changes on the source device). I've been trying to get it working but it doesn't seem to work for me and I am not sure if I am using the correct function for this.

I came across this and I believe this is a similar issue as I am having: https://github.com/pschatzmann/ESP32-A2DP/issues/41 

Many thanks 🙏

**Comment by pschatzmann** [comment, marked as answer]
Can you double check the Arduino ESP32 Version. You need to use a version > 2. The latest version is 2.0.2.
If you use this version and it is still not working then your phone might not support this function. In this case the set_volume() function is just applied on top of the volume provided by the phone... 

See https://github.com/pschatzmann/ESP32-A2DP/wiki/Controlling-your-Phone-with-AVRC-Commands

**Reply by AvishkaDon** [reply]
Many thanks! it works after upgrading :)

**Comment by kk-kk-k** [comment]
You can use esp-idf. On one hand, it is much more challenging and harder to learn, on the other hand it provides extra control (and maybe speed. Please note that speed is, when developing common applications, one of the last parameters considered. In many ue cases, it is lless important than etc. clarity, simplicity...). Here is link to esp-idf docs: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/bluetooth/esp_avrc.html
